diff --git a/node_modules/vitest/dist/index.d.ts b/node_modules/vitest/dist/index.d.ts
index 268e52c..f4d1e97 100644
--- a/node_modules/vitest/dist/index.d.ts
+++ b/node_modules/vitest/dist/index.d.ts
@@ -480,7 +480,7 @@ declare const assertType: AssertType;
 
 interface BrowserUI {
 	setCurrentFileId: (fileId: string) => void;
-	setIframeViewport: (width: number, height: number) => Promise<void>;
+	setIframeViewport: (width: number, height: number, scale?: number) => Promise<void>;
 }
 
 /** @deprecated import `TypeCheckRawErrorsMap` from `vitest/node` instead */
